integration/plugin/volumes/mounts_test.go

```diff
@@ -0,0 +1,58 @@
+package volumes
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/integration-cli/daemon"
+	"github.com/docker/docker/integration-cli/fixtures/plugin"
+	"github.com/gotestyourself/gotestyourself/assert"
+	"github.com/gotestyourself/gotestyourself/skip"
+)
+
+// TestPluginWithDevMounts tests very specific regression caused by mounts ordering
+// (sorted in the daemon). See #36698
+func TestPluginWithDevMounts(t *testing.T) {
+	skip.IfCondition(t, testEnv.IsRemoteDaemon(), "cannot run daemon when remote daemon")
```

Comment:
Just for my knowledge. How does the new test framework run on specific OSes? For example, how does it achieve the old `DaemonIsLinux` condition?

Reply:
This is handled here: https://github.com/moby/moby/pull/36711/files#diff-ead2a878e85c5d12126b9757bff4cf69R24

And yes I totally added that after you said something :)

Also removed this check (which was a copy/paste from logging) since we don't need it here.